src/marqo/s2_inference/s2_inference.py

```diff
@@ -67,14 +77,64 @@ def vectorise(model_name: str, content: Union[str, List[str]], model_properties:
         model_auth=model_auth
     )
 
+    if _marqo_inference_cache.is_enabled() and enable_cache:
+        if isinstance(content, str):
+            inference_cache_key = f"{model_cache_key}||{content}"
+            vectorised = _marqo_inference_cache.get(inference_cache_key)
+            if vectorised is None:
+                vectorised = _encode_without_cache(model_cache_key, content, normalize_embeddings, **kwargs)
+                _marqo_inference_cache[inference_cache_key] = vectorised[0]
+                return vectorised
+            else:
+                return _convert_cached_embeddings_to_output(vectorised)
+        elif isinstance(content, list):
+            contents_to_vectorise = []
+            cached_output: List[Tuple[int, List[float]]] = []
+
+            for loc, content_item in enumerate(content):
+                if not isinstance(content_item, str):
+                    contents_to_vectorise.append(content_item)
+                else:
+                    inference_cache_key = f"{model_cache_key}||{content_item}"
+                    vectorised = _marqo_inference_cache.get(inference_cache_key)
+                    if vectorised is None:
+                        contents_to_vectorise.append(content_item)
+                    else:
+                        cached_output.append((loc, vectorised))
+
+            if contents_to_vectorise:
+                vectorised_outputs: List[List[float]] = _encode_without_cache(model_cache_key, contents_to_vectorise,
+                                                                              normalize_embeddings, **kwargs)
+                for contents_to_vectorise_loc, vectorised_output in enumerate(vectorised_outputs):
+                    _marqo_inference_cache[f"{model_cache_key}||{contents_to_vectorise[contents_to_vectorise_loc]}"] = (
+                        vectorised_output)
+
+                for loc, cached_vector in cached_output:
+                    vectorised_outputs.insert(loc, cached_vector)
```

Comment:
Are we breaking the order here when some items in the list were cached and some were not?